publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/PublishUtils.java

```diff
@@ -56,16 +56,18 @@ public static MsgService getMessageService(String mp, MsgService msgServices[])
      * @param msgService the Messaging service.
      * @param json the eiffel event
      * @param userDomainSuffix is optional parameter, If user provide this it will add to the domainId.
+     * @param tag is optional parameter, If user provide this it will add to the tag if routingKey is not specified by the user.
+     * @param routingKey is optional parameter, If user provide this it will add to the routingKey otherwise generate routingKey based on message type.
      * @return routing key or returns "" if host, exchange and domainId not available.
     */
-    public static String getRoutingKey(MsgService msgService, JsonObject json, RMQHelper rmqHelper, String userDomainSuffix) {
+    public static String getRoutingKey(MsgService msgService, JsonObject json, RMQHelper rmqHelper, String userDomainSuffix, String tag, String routingKey) {
         String protocol = msgService.getServiceName();
         Boolean cliMode = Boolean.getBoolean(PropertiesConfig.CLI_MODE);
         RabbitMqProperties rabbitMqProperties = rmqHelper.rabbitMqPropertiesMap.get(protocol);
         String domainId = rabbitMqProperties.getDomainId();
         if (rabbitMqProperties != null && rabbitMqProperties.getExchangeName() != null && rabbitMqProperties.getHost() != null
```

Comment:
getroutingkey has nothing to do with rabbitmq propeties, could we move this out somehow